### ISSUE TITLE
Webpack v4.0 introduces plugin changes

### ISSUE THREAD

**Issue by pentatronix** [Va1/browser-sync-webpack-plugin#56]
This plugin may soon have issues, since the way the plugins are written has changed substantially:

See: Breaking changes for plugins/loaders
https://github.com/webpack/webpack/releases/tag/v4.0.0-beta.0

**Comment by malinushj** [comment]
@pentatronix Thanks for the info! I will keep my fork updated and do a pull request with the updates but I doubt that they'll get merged any time soon.

**Comment by Va1** [comment]
hi @pentatronix . thanks for the notice, i gonna update the plugin to support these breaking changes

**Comment by Va1** [comment]
@pentatronix @malinushj hi guys. with the release of webpack v4 i've reviewed the plugin code and did not find any difference between how it's written and how webpack suggests to write a plugin ([Plugin API](https://webpack.js.org/api/plugins/), [Writing a Plugin](https://webpack.js.org/contribute/writing-a-plugin/)). I have no project on Webpack 3 can painlessly migrate to check though. Have you?

**Comment by malinushj** [comment]
@Va1 @pentatronix  hello! didn't manage to have a look at the new docs but I do have some projects to migrate to webpack 4. I'll do that later today and come back with a response.

**Comment by malinushj** [comment]
@Va1 @pentatronix hi guys!
So Valentyn, you were right, the docs haven't been updated for v4 which is quite sad but I've managed to find the proper way to do it in [this repo](https://github.com/webpack/tapable).
Made #61 and #60.
Not sure if I proceeded correctly with the branching and tagging, feel free to correct me.
And let me know what you think.

As a side note, the adoption rate for the community isn't too great, in my opinion. I think they jumped the boat too fast, not updating the docs and not giving plugin developers more time to migrate... But overall I love v4! Awesome new features.
